File: src/test/java/MdxParsingTest.java

import com.intellij.testFramework.ParsingTestCase;
import org.intellij.plugin.mdx.lang.parse.MdxParserDefinition;

public class MdxParsingTest extends ParsingTestCase {
    public MdxParsingTest() {
        super("", "mdx", new MdxParserDefinition());
    }

    public void testParsingTestData() {
        doTest(true);
    }

    @Override
    protected String getTestDataPath() {
        return "src/test/testData";
    }

    @Override
    protected boolean skipSpaces() {
        return false;
    }

    @Override
    protected boolean includeRanges() {
        return true;
    }
}

File: src/test/java/MdxParsingTest.kt
import com.intellij.codeInspection.htmlInspections.HtmlUnknownAttributeInspection
import com.intellij.codeInspection.htmlInspections.HtmlUnknownBooleanAttributeInspection
import com.intellij.codeInspection.htmlInspections.HtmlUnknownTagInspection
import com.intellij.lang.javascript.dialects.ECMA6ParserDefinition
import com.intellij.testFramework.ParsingTestCase
import com.sixrr.inspectjs.validity.BadExpressionStatementJSInspection
import com.sixrr.inspectjs.validity.ThisExpressionReferencesGlobalObjectJSInspection
import org.intellij.plugin.mdx.js.MdxJSParserDefinition
import org.intellij.plugin.mdx.lang.parse.MdxParserDefinition
import org.intellij.plugins.markdown.lang.parser.MarkdownParserDefinition

class MdxParsingTest : ParsingTestCase("", "mdx", MdxParserDefinition(), MarkdownParserDefinition(), MdxJSParserDefinition(), ECMA6ParserDefinition()) {
    fun testParsingTestData() {
        doTest(true)
    }

    @Throws(Exception::class)
    public override fun setUp() {
        super.setUp()
    }

    override fun getTestDataPath(): String {
        return "src/test/testData/parsing"
    }

    override fun skipSpaces(): Boolean {
        return false
    }

    override fun includeRanges(): Boolean {
        return true
    }
}